### ISSUE TITLE
Auspice testing data (get-data)

### ISSUE THREAD

**Issue by jameshadfield** [nextstrain/auspice#1558]
## Background

This repo doesn't contain any datasets beyond some [very minimal examples to explain the dataset format](https://github.com/nextstrain/auspice/tree/master/examples). Instead we rely on the [get-data script](https://github.com/nextstrain/auspice/blob/master/scripts/get-data.sh) which downloads a slew of (nextstrain core) datasets. In times gone by, this was an accurate listing of all of our core datasets (and other sources didn't exist -- groups, community etc). This script is often run manually (e.g. `npm run get-data`) so you can have some data to play with, and heroku runs this during setup (`npm run heroku-postbuild`) which results in usable datasets in review apps.

## Shortcomings

A lot of auspice's functionality cannot be tested with the data here and two PRs in the last couple of weeks have highlighted this: https://github.com/nextstrain/auspice/issues/1463 and https://github.com/nextstrain/auspice/pull/1552. This means the heroku-review apps are not useful, and people have to manually checkout the auspice branch and obtain an appropriate dataset for testing.

## Proposal

We should make the `get-data` script obtain a useful set of testing datasets, preferably using timestamped datasets so we can ensure reproducibility. For PRs which need additional datasets to test, these should be added to the get-data script as part of the PR.

**Comment by victorlin** [comment]
Note that the PR review apps allow us to test on live nextstrain.org data

**Comment by jameshadfield** [comment]
> Note that the PR review apps allow us to test on live nextstrain.org data

That's true (and a big positive) but it doesn't help with local development and, some time in the future, actual tests within auspice.

I'll take a crack at this issue today - it seems simple on the surface!

**Comment by jameshadfield** [comment]
Resurrecting this issue, we now have:

* two (git-committed) datasets in `test/data`, both used for unit-tests. Ideally bugfixes would include a unit-test to prevent regressions, although I'm not sure how feasible this is.

* some very basic smoketests which currently depend on the `data/zika.json` dataset being provisioned.

* 8 "testing" narratives which are git-committed, but aren't leveraged for automated testing. They were added by me to help development of narratives but have largely fallen out of date because the reference datasets (in `data/`) which have been updated since the narrative was added.

* A `get-data.sh` script which downloads a mixture of date-stamped datasets and continuously updated datasets. 

I'm considering the following plan:

* Update all testing narratives (`narratives/`) to use nextstrain.org's `@YYYY-MM-DD` version ID or point to a specific git commit for community URLs. 
* Write a script to parse narratives, identify the required datasets, and download them to `data/`, munging their name such that they continue to work with narratives on localhost. This step (along with the previous one) will ensure the narrative test datasets don't fall out of sync with their datasets.
* Similarly parse the smoketest data, currently `urls.txt` but presumably better as a JSON / YAML, and download those files. They too should reference versioned datasets.
* Update our smoke-testing code to load the testing narratives and navigate through each slide, performing some basic check of accuracy. It's hard to see how to do it but the playright code / instructions could be written in each markdown slide to help prevent duplication.

**Comment by jameshadfield** [comment]
Auspice v3 (via PR #2967) has changed our approach here to only download/fetch timestamped datasets explicitly used for automated tests via `npm run fetch-test-data`
